File: TutorialCS/App_Code/Data/TimetableManager.cs
using System;
using System.Data;
using Data;

/// <summary>
/// Summary description for TimetableManager
/// </summary>
public class TimetableManager
{

    public static int UnmapBlock(TimeSpan start)
    {
        return start.Hours;
    }

    public static TimeSpan MapBlock(int id)
    {
        return TimeSpan.FromHours(id);
    }

    public static DateTime UnmapEventStart(DateTime start)
    {
        return UnmapEventStart(start.Date, start.TimeOfDay);
    }

    public static DateTime UnmapEventStart(DateTime day, TimeSpan time)
    {
        int block = UnmapBlock(time);
        return UnmapEventStart(day, block);
    }

    public static DateTime UnmapEventStart(DateTime day, int block)
    {
        DataRow dr = new DataManager().GetBlock(block);
        if (dr == null)
        {
            throw new Exception("Can't map event start (block not found).");
        }
        DateTime blockStart = (DateTime)dr["BlockStart"];
        return day.Date.Add(blockStart.TimeOfDay);
    }

    public static DateTime UnmapEventEnd(DateTime end)
    {
        return UnmapEventEnd(end.Date, end.TimeOfDay);
    }

    public static DateTime UnmapEventEnd(DateTime day, TimeSpan time)
    {
        int block = UnmapBlock(time) - 1;
        return UnmapEventEnd(day, block);
    }

    public static DateTime UnmapEventEnd(DateTime day, int block)
    {
        DataRow dr = new DataManager().GetBlock(block);
        if (dr == null)
        {
            throw new Exception("Can't map event start (block not found).");
        }
        DateTime blockEnd = (DateTime)dr["BlockEnd"];
        return day.Date.Add(blockEnd.TimeOfDay);
    }

}

File: TutorialVB/App_Code/Data/TimetableManager.vb
Imports System
Imports System.Data
Imports Data

''' <summary>
''' Summary description for TimetableManager
''' </summary>
Public Class TimetableManager

	Public Shared Function UnmapBlock(ByVal start As TimeSpan) As Integer
		Return start.Hours
	End Function

	Public Shared Function MapBlock(ByVal id As Integer) As TimeSpan
		Return TimeSpan.FromHours(id)
	End Function

	Public Shared Function UnmapEventStart(ByVal start As Date) As Date
		Return UnmapEventStart(start.Date, start.TimeOfDay)
	End Function

	Public Shared Function UnmapEventStart(ByVal day As Date, ByVal time As TimeSpan) As Date
		Dim block As Integer = UnmapBlock(time)
		Return UnmapEventStart(day, block)
	End Function

	Public Shared Function UnmapEventStart(ByVal day As Date, ByVal block As Integer) As Date
		Dim dr As DataRow = (New DataManager()).GetBlock(block)
		If dr Is Nothing Then
			Throw New Exception("Can't map event start (block not found).")
		End If
		Dim blockStart As Date = CDate(dr("BlockStart"))
		Return day.Date.Add(blockStart.TimeOfDay)
	End Function

	Public Shared Function UnmapEventEnd(ByVal [end] As Date) As Date
		Return UnmapEventEnd([end].Date, [end].TimeOfDay)
	End Function

	Public Shared Function UnmapEventEnd(ByVal day As Date, ByVal time As TimeSpan) As Date
		Dim block As Integer = UnmapBlock(time) - 1
		Return UnmapEventEnd(day, block)
	End Function

	Public Shared Function UnmapEventEnd(ByVal day As Date, ByVal block As Integer) As Date
		Dim dr As DataRow = (New DataManager()).GetBlock(block)
		If dr Is Nothing Then
			Throw New Exception("Can't map event start (block not found).")
		End If
		Dim blockEnd As Date = CDate(dr("BlockEnd"))
		Return day.Date.Add(blockEnd.TimeOfDay)
	End Function

End Class